lib/configuration.js

```diff
@@ -174,7 +175,19 @@ var Configuration = function(givenConfig, logger) {
  * @returns {Undefined} - does not return anything
  */
 Configuration.prototype._checkLocalServiceContext = function() {
-  var gaeModuleName = env.GAE_MODULE_NAME;
+  // Note: The GAE_MODULE_NAME environment variable is set on GCP.
+  //       If the code is, in particular, running on GCF, then the 
+  //       FUNCTION_NAME environment variable is set.
+  //
+  // To determine the service name to use:
+  //   If the user specified a service name it should be used, otherwise 
+  //   if the FUNCTION_NAME environment variable is set (indicating that the 
+  //   code is running on GCF) then the FUNCTION_NAME value should be used as 
+  //   the service name.  If neither of these conditions are true, the 
+  //   value of the GAE_MODULE_NAME environment variable should be used as the 
+  //   service name.
+  var funcName = env.FUNCTION_NAME;
+  var gaeModuleName = isUndefined(funcName) ? env.GAE_MODULE_NAME : funcName;
   var gaeModuleVer = env.GAE_MODULE_VERSION;
```

Comment:
Does GCF have a concept of function version? Intuitively `GAE_MODULE_VERSION` should be used iff we also are going to use `GAE_MODULE_NAME` as the service name.